<launch>
  <param name="/use_sim_time" value="true"/>
 <arg name="bag"> </arg>
  <node pkg="rviz" type="rviz" name="rviz" args="-d$(find diffdrive_calibrator)/data/calib_view.vcg" />


  <node name="rosbag" type="play" pkg="rosbag" args="$(find diffdrive_calibrator)/data/$(arg bag) -k  --clock "/>
 
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser" 
    args="0.0 0.0 0.0 0 0 0 /gt_frame /base_laser_link 40" />
	  
	  <node pkg="tf" type="static_transform_publisher" name="world_to_odom" 
    args="0.0 0.0 0.0 0 0 0 /world /odom 40" />

	<node pkg="navi_driver" type="navi_driver.py" name="navi_driver" output="screen">
		<rosparam commmand="load" file="$(find navi_driver)/diff_config.yaml" />
		<remap from="motor_cmd" to="/drive/motor_cmd"/>
		<remap from="encoder" to="/drive/encoder"/>
		<remap from="cmd_vel" to="/drive/cmd_vel"/>
		<remap from="odom" to="/drive/odom"/>
	</node>
<!--
<node pkg="canonical_scan_matcher" type="csm_node" name="csm_node" output="screen">
		<remap from="scan" to="base_laser/scan"/>
		<param name="max_correspondence_dist" value="0.25"/>
		<param name="orientation_neighbourhood" value="20"/>
		<param name="outliers_maxPerc" value="0.75"/>
		<param name="max_angular_correction_deg" value="10"/>
		</node>
-->

 <node pkg="polar_scan_matcher" type="psm_node" name="psm_node" output="screen">
  	<remap from="scan" to="base_laser/scan"/>
    <param name="max_error" value="0.20"/>
    <param name="search_window" value="100"/>
    <param name="stop_condition" value="0.005"/>
    <param name="max_iterations" value="30"/>
    <param name="base_frame" value="gt_frame"/>
  </node>
	
</launch>

